### ISSUE TITLE
Random Forest regression issue: the regression loss never decreases

### ISSUE THREAD

**Issue by stephenzwj** [Microsoft/LightGBM#1320]
Hi,
I've been trying to use RF with regression objective in LightGBM for python, but the loss value never decreases. Even if I tune the parameters it never decreases. Is Random Forest implemented for regression in LightGBM for python? Can anyone shares the settings with me?

## Environment info
Operating System: Windows 10
CPU: INTEL I5
C++/Python/R version: Python

## Error Message:
There is no error, but the regression loss function never decreases when setting the boosting as rf
## Reproducible examples
        param = {
            'boosting_type': 'rf',
            'objective':'regression_l2',
            'num_leaves':147,
            'learning_rate': 0.1,
            'metric':'mse',
            'max_depth': -1
             'verbose': 1,
            'device':'cpu',
            'num_threads': 3
             'tree_learner':'feature',
             'reg_alpha': 0.1
             'reg_lambda':0.1
             'bagging_fraction':0.621,
            'feature_fraction':0.621,
            'bagging_freq':1,
            'n_estimators':4000,
            'min_data_in_leaf':10,
            'min_sum_hessian_in_leaf':0}

## Steps to reproduce

1. Running lightGBM by setting the boosting type as rf and the application as regression

**Comment by guolinke** [comment]
how many data you use ?  what the mean of your regression targets ?

**Comment by stephenzwj** [comment]
I use 42984 samples and there are 103 features in each sample. The mean of my regression targets is 147.6040. Samples and targets are all in float type.

**Comment by mayer79** [comment]
What happens if you set the learning rate to 1?

**Comment by guolinke** [comment]
@stephenzwj I think the problem can be solved if you covert the mean of regression targets to 0 (by minus 147.6040). The reason is, the output of tree is limit to (-100,100). 
I will change this hard constraint to a configurable parameter soon.

**Comment by guolinke** [comment]
refer to #1322

**Comment by stephenzwj** [comment]
@guolinke Yes, it is working now.

**Comment by stephenzwj** [comment]
@mayer79 Varying the learning rate from 1 to 1e-5, there is no difference. It seems the problem is the output constraint, which havs been mentioned be guolinke.
